ste/securityInfoPersistenceManager.go

```diff
@@ -2,127 +2,87 @@ package ste
-		serviceURLs: make(map[string]azfile.ServiceURL),
-		ctx:         ctx,
+		sipmMu: &sync.Mutex{},
+		cache:  lru.New(3000), // Assuming all entries are around 9kb, this would use around 30MB.
+		ctx:    ctx,
 	}
 }
 
-func (sipm *securityInfoPersistenceManager) PutServiceURL(serviceURL azfile.ServiceURL) {
-	fURLParts := common.NewGenericResourceURLParts(serviceURL.URL(), common.ELocation.File())
-
-	sipm.sipmMu.Lock()
-	defer sipm.sipmMu.Unlock()
-
-	sipm.serviceURLs[fURLParts.GetAccountName()] = serviceURL
-}
-
 // Technically, yes, GetSDDLFromID can be used in conjunction with PutSDDL.
 // Being realistic though, GetSDDLFromID will only be called when downloading,
 // and PutSDDL will only be called when uploading/doing S2S.
-func (sipm *securityInfoPersistenceManager) PutSDDL(acctName, shareName, sddlString string, p pipeline.Pipeline) (string, error) {
-	sipm.sipmMu.Lock()
-	defer sipm.sipmMu.Unlock()
+func (sipm *securityInfoPersistenceManager) PutSDDL(sddlString string, shareURL azfile.ShareURL) (string, error) {
+	fileURLParts := azfile.NewFileURLParts(shareURL.URL())
+	fileURLParts.DirectoryOrFilePath = ""
+	rawfURL := fileURLParts.URL()
+
+	sddlKey := rawfURL.String() + "|SDDL|" + sddlString
 
 	// First, let's check the cache for a hit or miss.
 	// These IDs are per share, so we use a share-unique key.
 	// The SDDL string will be consistent from a local source.
-	id, ok := sipm.cache.Get(acctName + "|SHARE|" + shareName + "|SDDL|" + sddlString)
-
-	fmt.Println(acctName + "|SHARE|" + shareName + "|SDDL|" + sddlString)
+	id, ok := sipm.cache.Get(sddlKey)
 
 	if ok {
-		fmt.Println("cache HIT!")
 		return id.(string), nil
 	}
 
-	parsedSDDL, err := sddl.ParseSDDL(sddlString)
-
-	if err != nil {
-		return "", err
-	}
-
-	// No extra work is needed to make SIDs portable, because SID.String() calls SID.ToPortable() for us.
-	upSDDL := parsedSDDL.PortableString()
-
-	serviceURL, ok := sipm.serviceURLs[acctName]
-
-	if !ok {
-		return "", fmt.Errorf("service URL wasn't found for account %s", acctName)
-	}
-
-	serviceURL = serviceURL.WithPipeline(p)
-
-	shareURL := serviceURL.NewShareURL(shareName)
-
-	cResp, err := shareURL.CreatePermission(sipm.ctx, upSDDL)
+	cResp, err := shareURL.CreatePermission(sipm.ctx, sddlString)
 
 	if err != nil {
 		return "", err
 	}
 
-	key := cResp.FilePermissionKey()
+	permKey := cResp.FilePermissionKey()
 
-	sipm.cache.Add(acctName+"|SHARE|"+shareName+"|SDDL|"+sddlString, key)
+	sipm.cache.Add(sddlKey, permKey)
 
-	return key, nil
+	return permKey, nil
 }
 
-func (sipm *securityInfoPersistenceManager) GetSDDLFromID(acctName, shareName, id string, p pipeline.Pipeline) (string, error) {
-	sipm.sipmMu.Lock()
-	defer sipm.sipmMu.Unlock()
+func (sipm *securityInfoPersistenceManager) GetSDDLFromID(id string, shareURL azfile.ShareURL) (string, error) {
+	fileURLParts := azfile.NewFileURLParts(shareURL.URL())
+	fileURLParts.DirectoryOrFilePath = ""
+	rawfURL := fileURLParts.URL()
```

Comment:
I'm a bit lost here. Haven't you already got the share URL? Didn't the caller give it to you?  So I'm lost what what the additional manipulation here (and in PutSDDL) is for.  Unless, I guess, you want to remove all the query string parts here, since they don't add value to the key, but do add length.  That might be worth doing, but I don't think that's what _this_ code is doing.  So I'm a little lost. Help! ;-)

Reply:
I may've been a little tired when I was writing this.

For some reason, I didn't think about already having the path stripped. But yes, I do want the SAS stripped.

Here's a thought for the future, so we aren't perpetually passing around pre-signed URLs: Perhaps we could consider having a pipeline policy to apply the SAS for us?

Reply:
(This has been resolved, but leaving the convo open for the last idea)

Reply:
Interesting suggestion.  I always feel a bit cautious about security-related changes.  But that doesn't mean we shouldn't do it (one day). Let's discuss with @zezha-msft one day.

Reply:
Resolving the thread, on the assumption that you'll follow up with me and Ze separately later :-)